privacyscore/backend/management/commands/scanfromfile.py

```diff
@@ -27,24 +27,28 @@ class Command(BaseCommand):
 
     def add_arguments(self, parser):
         parser.add_argument('file_path')
+        parser.add_argument('scan_list')
         parser.add_argument('-s', '--sleep-between-scans', type=float, default=0)
 
     def handle(self, *args, **options):
         if not os.path.isfile(options['file_path']):
             raise ValueError('file does not exist!')
         self.stdout.write('Reading from file {}'.format(options['file_path']))
 
-        start = timezone.now()
+        scan_list = ScanList.objects.get_or_create(name=options['scan_list'])[0]
```

Comment:
I'm wondering whether it's smarter to always have new lists created in order to avoid destroying existing data. That would probably also make the code a bit simpler. Although we'd probably still want to give the list a name.
Do you have a use case for modifying an existing list?

Reply:
Discussions in the group chat brought up the idea to have two options:
- A `--rescan` switch that takes a list ID instead of a CSV to initiate a rescan of the list (with the same rate limiting as scanning a new list)
- In case it is not a rescan, check if a list with the name already exists and, if so, refuse to proceed.

This way, we avoid accidental edits to the existing lists, which are (for now) supposed to remain static.

Reply:
I have added a rescan option as suggested by @malexmave

Reply:
I don't see how you've added a rescan option as suggested. You're proposing to make it take the name of the list as an argument. Suggested was to take the ID.
I think the name makes it awkward to use. Take the currently existing https://privacyscore.org/list/59 for example.  How would I use this command now to trigger a re-scan?  `manage.py scanfromfile 'Predatory Open Access Publishers (Beall's List)'`?  That's very fragile.  Also it's very counter intuitive to have a re-scan triggered with a command named "scanfromfile".

Reply:
The rescan option is already available for the lists which are public. For example you can go to https://privacyscore.org/list/26/ and re-scan from there. So this script specifically targets the rescan of the list which is not visible, as well as fulfils the requirement for my thesis data collection. If you want, I can take the ID as input instead of name but otherwise I think it is fine.